website/orders/api/v1/views.py

```diff
@@ -0,0 +1,393 @@
+import datetime
+
+import pytz
+from django.db.models import Q
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.exceptions import PermissionDenied, ParseError, ValidationError
+from rest_framework.generics import (
+    ListCreateAPIView,
+    ListAPIView,
+    RetrieveUpdateAPIView,
+    RetrieveDestroyAPIView,
+)
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from orders import services
+from orders.api.v1.permissions import HasPermissionOnObject
+from orders.api.v1.serializers import OrderSerializer, ShiftSerializer, ProductSerializer
+from orders.exceptions import OrderException
+from orders.models import Order, Shift, Product
+from orders.services import Cart, place_orders, increase_shift_time, increase_shift_capacity
+from tosti import settings
+
+
+class CartOrderAPIView(APIView):
+    """
+    Cart Order API View.
+
+    Permission required: shift.can_order_in_venue
+
+    Use this API endpoint to order a list of Products in one go. The list of Products should be set as an array of
+    Product id's in the "cart" POST parameter.
+    """
+
+    def _extract_cart(self):
+        """
+        Extract the cart items from the POST data.
+
+        :return: a Cart object with the cart items as specified in the cart in the POST data. Raises a ValidationError
+        when there is not cart in the POST data. Raises a ParseError when the cart can not be parsed.
+        """
+        cart_as_id_list = self.request.data.get("cart", None)
+        if cart_as_id_list is None:
+            raise ValidationError
+
+        try:
+            return Cart.from_list(cart_as_id_list)
+        except ValueError:
+            raise ParseError
+
+    def get_object(self):
+        """
+        Get the Shift the user wants to add the orders to.
+
+        :return: a Shift object
+        """
+        return self.kwargs.get("shift")
+
+    def post(self, request, **kwargs):
+        """
+        Create multiple Orders in one go.
+
+        API endpoint for creating multiple orders in one go (handling of a cart).
+        A "cart" POST parameter must be specified including the ID's of the Products the user wants to order.
+        Permission required: can_order_in_venue
+        :param request: the request
+        :param kwargs: keyword arguments
+        :return: a 201 response code if all orders in the cart were successfully processed. Raises a ValidationError
```

Comment:
We have pydoc `:raises:`

Reply:
I have removed most of them as they were displayed on the SwaggerUI page (and that was ugly).